controller/workitem.go

```diff
@@ -561,6 +563,199 @@ func getVersion(version interface{}) (int, error) {
+	for i := 0; i < len(wis); i++ {
+		// iterate over the WIs, converting them one by one.
+		fieldLabels, _, fieldValues, err := ConvertWorkItemToStringValue(ctx, app, &uuidStringCache, wits[i], wis[i])
+		if err != nil {
+			return "", errs.Wrapf(err, "failed to convert work item to CSV: %s", wis[i].ID)
+		}
+		// create a new line, yet without potential new columns.
+		csvLine := make([]string, len(csvGrid[0]))
+		// add new new (empty) line to the grid.
+		csvGrid = append(csvGrid, csvLine)
+		// now, mangle up the column configuration
+		for j := 0; j < len(fieldLabels); j++ {
+			// iterate over the field keys, mapping them to the resulting grid
+			// the CSV key is unique and contains the WIT's name and the fieldLabel
+			// there might be collisions due to WITs names and/or fieldKeys are not
+			// required to be unique, but for the usecase, if there is an equally
+			// named WIT/fieldKey in a template, it is highly probable that this
+			// is intended to go into the same column.
+			csvFieldKey := fieldLabels[j]
+			// now check if the current key is a new key.
+			_, ok := columnMap[csvFieldKey]
+			if !ok {
+				// this is a new key, create a new columnMap entry.
+				columnMap[csvFieldKey] = len(csvGrid[0])
+				// add the new header entry.
+				csvGrid[0] = append(csvGrid[0], csvFieldKey)
+				// append empty values to all existing "lines" for the new column.
+				for k := 1; k < len(csvGrid); k++ {
+					csvGrid[k] = append(csvGrid[k], "")
+				}
+			}
+			// now we are ready to put the value where is belongs in the grid.
+			gridRowIdx := len(csvGrid) - 1 // last line.
+			gridColumnIdx := columnMap[csvFieldKey]
+			csvGrid[gridRowIdx][gridColumnIdx] = fieldValues[j]
+		}
+	}
+	// create CSV from the [][]string
+	buf := new(bytes.Buffer)
+	w := csv.NewWriter(buf)
+	err := w.WriteAll(csvGrid)
+	if err != nil {
+		return "", errs.Wrapf(err, "failed to serialize to CSV format")
+	}
+	// done, return serialized result.
+	return buf.String(), nil
+}
+
+// ConvertWorkItemToStringValue is responsible for converting given WorkItem model object into a
+// string slice; it returns the field names, the field keys and the string converted values for the work item
+func ConvertWorkItemToStringValue(ctx context.Context, app application.Application, uuidStringCache *map[string]string, wit workitem.WorkItemType, wi workitem.WorkItem) ([]string, []string, []string, error) {
+	fieldNames := []string{}
+	fieldKeys := []string{}
+	fieldValues := []string{}
+	for fieldKey, fieldDefinition := range wit.Fields {
+		// convert the value to a string for the CSV
+		fieldNames = append(fieldNames, fieldDefinition.Label)
+		fieldKeys = append(fieldNames, fieldKey)
+		fieldValueGeneric := wi.Fields[fieldKey]
+		fieldType := fieldDefinition.Type
+		fieldValueStr, err := fieldType.ConvertToStringSlice(fieldValueGeneric)
+		if err != nil {
+			return nil, nil, nil, errs.Wrapf(err, "failed to convert simple type value to string for field key: %s", fieldKey)
```

Comment:
This isn't `simple type` specific. Shouldn't it be `failed to convert type value to string....` ?

Reply:
Correct.

Reply:
Fixed 7ae9140.